### DISCUSSION TITLE
v0.3.0 - Jam Protection

### DISCUSSION THREAD

**Discussion by smarthausleben** [smarthausleben/ioBroker.wamo#23] [Announcements]
This **_Jam Protection release_** adds a scheduled move of the main valve which can be activated and scheduled in the adapter settings area (Tab: Tasks)

Be verry careful with this functionality because it will **_close_** and **_open_** the main valve in order to prevent it to get stuck in open position and therefore will not close in case of an leakage.

The **_Jam Protection_** can be planed using a CRON schedule which will also be configured in the adapter settings **_Tasks_** tab.
Be verry careful there as well, because if you schedule a jam protection for example every 1 Minute you are having big trouble because your main valve will close and open every minute!

During the **_Jam Protection_** activity, no regular states (Valve, Alarms etc.) will be updated in order to prevent false trigger you may have set on of those states.

If the main valve is already in **_closed_** position, the **_Jam Protection_** activity will not be performed in order to prevent opening the main valve.
If there is water consumption at start of **_Jam Protection_** the action will be delayed for 1 minute. After 10 faild attempts (water still flowing)  **_Jam Protection_** will be aborted.

### Disclaimer / Warning
If during the **_Jam Protection_** activity communication to the device gets lost or the WAMO adapter or ioBroker itself crashes or will be stopped, the main valve will stay in the last commanded position! This means if something gets wrong, the valve could be closed and needs to be opened by yourself using the related app or the button on the device itself.

<hr /><em>This discussion was created from the release <a href='https://github.com/smarthausleben/ioBroker.wamo/releases/tag/v0.3.0'>Jam Protection</a>.</em>

**Comment by smarthausleben** [comment]
The new Version is showing up in the ioBroker **_Adapter_** Panel 😅 **_(beta/latest only!)_**
![image](https://user-images.githubusercontent.com/101345409/230037907-1faff8d8-c942-41b3-ab89-269df183e5b3.png)

**Comment by pustelbaer** [comment]
in my adapter browser wamo is not listed for the stable repository. Only with beta/latest repo, the 0.2.13 appears:
![wamo](https://github.com/smarthausleben/ioBroker.wamo/assets/35026034/b4705f36-7aab-4582-86dd-d7d4334846a7)

**Reply by smarthausleben** [reply]
Sorry, but this is absolutely correct because I didn’t went through the administrative process to get the adapter into the stable repositories list yet. 🙄 But you should definitely see version **0.3.0** if you switch to beta/latest.
Currently I’m working on the implementation of the SafeTech Floor Sensors. Maybe I give it a try if this part is completed.
So in order to get the new release, you have to switch to beta/latest. After the update you may switch back to stable again.

**Reply by pustelbaer** [reply]
Hmm, when I switch to the beta repository, only 0.2.13 is displayed. In the release repo the adapter is not listed at all.
So I'll stay at 0.2.13 for now.

**Reply by smarthausleben** [reply]
This is strange because on Github I see the release and on npm as well? I have to aske somebody from the ioBroker team. Maybe I have to do something else there. But usually in the beta repositories pushing a new version to nmp should do the trick.

**Reply by smarthausleben** [reply]
> Hmm, when I switch to the beta repository, only 0.2.13 is displayed. In the release repo the adapter is not listed at all. So I'll stay at 0.2.13 for now.

Which version of the `Admin` adapter are you running on your ioBroker system? Requirement for wamo version `0.3.0` is `Admin 5.x` So if you are running a version below 5 this may be the reason you don't see 0.3.0

**Comment by pustelbaer** [comment]
I'm running Admin 6.3.5

**Reply by smarthausleben** [reply]
To be honest, I don't have any idea whats the problem on your ioBroker installation. Maybe your Nodes version? You shold be on  >=16.x.

ioBroker.dev shows npm 0.3.0
![image](https://github.com/smarthausleben/ioBroker.wamo/assets/101345409/1c96f47e-c840-4dfb-a546-8a0f355a7678)

 and if I check the statistics, the number of installed 0.3.0 versions is growing as well.
![image](https://github.com/smarthausleben/ioBroker.wamo/assets/101345409/d07a1ca2-79a0-45c4-aeed-a28c23297089)

Maybe you should try to install it direct from NMP. If you install it from there, which version do you get?
![image](https://github.com/smarthausleben/ioBroker.wamo/assets/101345409/ba84f2f2-a195-4a9e-bc61-2699d92331f2)

**Comment by pustelbaer** [comment]
Thanks, I've installed / updated via URL. from github.

Now with 0.3.0 my log is overflowing with query timer messages: 

![grafik](https://github.com/smarthausleben/ioBroker.wamo/assets/35026034/52f04c16-9ff3-487c-af30-1c67ff224c0d)

while I have not checked any of the additional logging messages in the extended option settings.

**Reply by smarthausleben** [reply]
Currently there is no option to hide all log.info messages. Maybe I will find some time to add this option into the settings before the next release.

**Reply by smarthausleben** [reply]
New option to hide **Trigger** logging messages is implemented and will be available in next release.
Related Issue: #26 

![image](https://github.com/smarthausleben/ioBroker.wamo/assets/101345409/168e92bb-5d44-41ae-ab6a-3a76b39958c9)
